include/fi_proto.h

```diff
@@ -90,10 +91,8 @@ struct ofi_ctrl_hdr {
 	uint8_t			type;
 	uint16_t		seg_size;
 	uint32_t		seg_no;
-	union {
-		uint64_t	conn_id;
-		uint64_t	msg_id;
-	};
+	uint64_t		conn_id;
+	uint64_t		msg_id;
```

Comment:
Note that this changes the size of the structure.  Other structures may need to be adjusted accordingly.